### DISCUSSION TITLE
Ability to obfuscate or prevent saving of IP addresses.

### DISCUSSION THREAD

**Discussion by OoLunar** [EssentialsX/Essentials#4880] [Ideas and Feature Suggestions]
Hello!

I'm a server owner for a small group of friends and I'd like to hide their IP addresses in case I ever accidentally leak them when sending the plugins folder. I understand well that most of the time, IP addresses can't do much harm without the end-user doing extra action. I also understand that it's not difficult to get the IP address of another player. Finally, I understand that the `banned-ips.json` file uses IP addresses.

I am not suggesting doing away with IP addresses, I'm suggesting storing them a bit more securely than in a raw `.yml` file.  This is a feature I'd like to see added so that I can use it out of respect for my friend's privacy, and to slightly increase security.

**Comment by Creeperz653** [comment]


**Reply by OoLunar** [reply]
I'm referencing the files that Essentials generate, such as the files located in `plugins/Essentials/userdata`. Essentials has no control over the `banned-ips.json` file because this is a file that the vanilla server uses.
